### DISCUSSION TITLE
Which MS Visual Studio and Qt Version are required to build Sandboxie Plus?

### DISCUSSION THREAD

**Discussion by engr-m-ayoub** [sandboxie-plus/Sandboxie#1089] [Documents and notes]
Hi,
I've tried building sandboxie plus with MSVS2019 and Qt 5.12.11 with msvc2017 kit but facing some qt method issues such as:

Sandboxie-master\SandboxiePlus\QSbieAPI\SbieAPI.cpp:1254: error: C2661: 'QMap<quint32,CBoxedProcessPtr>::insert': no overloaded function takes 1 arguments

Sandboxie-master\SandboxiePlus\SandMan\SandMan.cpp:1699: error: C2039: 'screen': is not a member of 'CSandMan'
Sandboxie-master\SandboxiePlus\SandMan\SandMan.h(27): note: see declaration of 'CSandMan'

Which version of MSVC/Qt is required to build?
Is there any build guide available ? 

Thank you in advance.

**Comment by engr-m-ayoub** [comment]


**Reply by DavidXanatos** [reply]
See the following references:
- https://github.com/sandboxie-plus/Sandboxie/blob/master/SandboxiePlus/ReadMe.md
- https://github.com/sandboxie-plus/Sandboxie/issues?q=%20label%3A%22Type%3A%20Build%22

**Reply by engr-m-ayoub** [reply]
Thank you @DavidXanatos . It worked.

**Reply by engr-m-ayoub** [reply]
@DavidXanatos I need your help again in building sanboxie. I built sandboxie-plus but it require sandboxie as well.
When I tried building with 10.0 SDK/WDK VS2019, I'm facing many issues for missing symbols and headers.

Can you please share your Project configurations which build it smoothly? I've tried all toolset options, as shown below, and get different error with each.

![image](https://user-images.githubusercontent.com/44095200/128996556-b8a0ef92-bc31-423f-8d18-a680c03343a3.png)

**Reply by isaak654** [reply]
One of the errors is related to missing Spectre libraries, and David replied [here](https://github.com/sandboxie-plus/Sandboxie/issues/1086#issuecomment-894835074) that they are irrelevant for this project. So I assume it's the same for the others.

**Reply by engr-m-ayoub** [reply]
@isaak654  Thank you for responding. I've disabled /QSpectre but still facing issues for missing header and libs like ntifs.h
Will appreciate if you can help with toolset selection that works to build

**Reply by isaak654** [reply]
Try to replace the `#include <ntifs.h>` references with `#include "ntifs.h"`:
https://github.com/sandboxie-plus/Sandboxie/search?q=ntifs

**Reply by DavidXanatos** [reply]
I'm surprised you have so many issues,
when I set up a new dev VM like for win 11 recently I just check out the sources install VS 2019 and the most recent WDK and that's it.
That should build.
Although one thing to keep in myind is that the driver iirc never built as debug for 32 and may be 64 bit hence i always build it release also for testing

**Reply by DavidXanatos** [reply]
i think ntifs.h missing indicates you have not installed WDK.... aaaaAAAA hahaha!
that may be it you need to install the VS2019 WDK extension as well, the wdk installer offers this but if VS was open during instalation that fails, or you may have unselected it.
Withut teh extension VS does not seam to knwo where to look for the includes
